report-generator/src/main/java/com/teamscale/report/jacoco/JaCoCoXmlReportGenerator.java

```diff
@@ -58,27 +61,26 @@ public JaCoCoXmlReportGenerator(List<File> codeDirectoriesOrArchives,
 	 *
 	 * @return The file object of for the converted report or null if it could not be created
 	 */
-	public File convert(Dump dump, String filePath) throws IOException {
+	public CoverageFile convert(Dump dump, Path filePath) throws IOException {
 		File outputFile = createOutputFile(filePath);
-		if (outputFile == null) {
-			return null;
-		}
 		FileOutputStream output = new FileOutputStream(outputFile);
 		convertToReport(output, dump);
-		return outputFile;
+		return new CoverageFile(outputFile);
 	}
 
-	private File createOutputFile(String filePath) {
-		File outputFile = new File(filePath);
+	private File createOutputFile(Path filePath) throws IOException {
+		File outputFile = filePath.toFile();
 
 		try {
-			boolean fileCreated = outputFile.createNewFile();
-			if (!fileCreated) {
-				return null;
-			}
+			Files.createFile(outputFile.toPath());
+		} catch (FileAlreadyExistsException e) {
+			// Clear file contents
+			new FileOutputStream(outputFile).close();
 		} catch (IOException e) {
-			logger.error("Could not create file " + outputFile.getAbsolutePath() + ". ");
-			return null;
+			// Add additional info to the IOException
```

Comment:
is this a TODO comment for you? if not, please either remove or explain as it's confusing for me what you mean with this